drkafka/src/main/java/com/pinterest/doctorkafka/KafkaCluster.java

```diff
@@ -254,35 +287,129 @@ public BrokerStats getLatestBrokerStats(int brokerId) {
     List<KafkaBroker> unusableBrokers = new ArrayList<>();
     for (int oosBrokerId : oosReplica.outOfSyncBrokers) {
       // we will get the broker with the least network usage
-      KafkaBroker leastUsedBroker = brokerQueue.poll();
-      while (leastUsedBroker != null && oosReplica.replicaBrokers.contains(leastUsedBroker.id())) {
-        unusableBrokers.add(leastUsedBroker);
-        leastUsedBroker = brokerQueue.poll();
-      }
-      if (leastUsedBroker == null) {
-        LOG.error("Failed to find a usable broker for fixing {}:{}", oosReplica, oosBrokerId);
-        success = false;
-      } else {
-        LOG.info("LeastUsedBroker for replacing {} : {}", oosBrokerId, leastUsedBroker.id());
-        success &= leastUsedBroker.reserveInBoundBandwidth(topicPartition, inBoundReq);
-        if (preferredBroker == oosBrokerId) {
-          success &= leastUsedBroker.reserveOutBoundBandwidth(topicPartition, outBoundReq);
-        }
-        if (success) {
-          result.put(oosBrokerId, leastUsedBroker);
-          // the broker should not be used again for this topic partition.
-          unusableBrokers.add(leastUsedBroker);
-        } else {
-          LOG.error("Failed to allocate resource to replace {}:{}", oosReplica, oosBrokerId);
-          success = false;
-        }
-      }
+      success &= findNextBrokerForOosReplica(
+          brokerQueue,
+          unusableBrokers,
+          oosReplica.replicaBrokers,
+          result,
+          oosBrokerId,
+          topicPartition,
+          inBoundReq,
+          outBoundReq,
+          preferredBroker
+      );
     }
     // push the brokers back to brokerQueue to keep invariant true
     brokerQueue.addAll(unusableBrokers);
     return success ? result : null;
   }
 
+  /**
+   * Similar to getAlternativeBrokers, but locality aware
```

Comment:
a general  question:  when `broker.rack` is not set on the kafka cluster side,  kafkastats will have `null` for `rackId`.  In that case, if the user set `locality_awareness.enabled` to `true` in the setting, what will be the expected behavior?  can we add explanation about this in the comment?